src/context/TourContext/TourContextProvider.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import Joyride, { type CallBackProps, STATUS, type Step } from 'react-joyride';
+
+import TourTooltip from '~common/Tours/TourTooltip.tsx';
+import { useActionSidebarContext } from '~context/ActionSidebarContext/ActionSidebarContext.ts';
+
+import { TourContext, type TourContextValue } from './TourContext.ts';
+
+interface TourContextProviderProps {
+  children: React.ReactNode;
+}
+
+const TourContextProvider: React.FC<TourContextProviderProps> = ({
+  children,
+}) => {
+  const [run, setRun] = useState(false);
+  const [steps, setSteps] = useState<Step[]>([]);
+  const [stepIndex, setStepIndex] = useState(0);
+  const {
+    actionSidebarToggle: [isSidebarOpen, { toggleOn: toggleActionSidebarOn }],
+  } = useActionSidebarContext();
+
+  const startTour = useCallback((tourSteps: Step[]) => {
+    const stepsWithData = tourSteps.map((step) => {
+      const data = step.data || {};
+      return {
+        ...step,
+        data: {
+          ...data,
+          triggerIdentifier: data.triggerIdentifier,
+          triggerPayload: data.triggerPayload,
+          icon: data.icon,
+          image: data.image,
+          imageAlt: data.imageAlt,
+          title: step.title,
+        },
+      };
+    });
```

Comment:
Do we need this mapping? It just re-adds properties that already exist because of the spread, apart from `data.title`. At the same time `data.title` doesn't seem to be used anywhere in our codebase (but I'm not sure if it's maybe needed by react-joyride).

Reply:
Not anymore, good pickup!